{
  try {
    if (myClosed) {
      command.cancel();
    }
 else {
      command.invoke();
    }
  }
 catch (  Throwable t) {
    LOG.error("Command " + command + " threw an exception.",t);
  }
}
